go/src/lib.rs

```diff
@@ -379,6 +394,70 @@ pub unsafe extern "C" fn command(
                 command_response.float_value = num;
                 Ok(Some(command_response))
             }
+            Value::Boolean(bool) => {
+                command_response.bool_value = bool;
+                Ok(Some(command_response))
+            }
+            Value::Array(array) => {
+                let len = array.len();
+                let mut vec_elements_len: Vec<c_long> = Vec::<c_long>::with_capacity(len);
+                let vec = array
+                    .iter()
+                    .map(|value| {
+                        if *value == Value::Nil {
+                            vec_elements_len.push(0);
+                            return std::ptr::null_mut();
+                        }
+                        let res = <String>::from_redis_value(value)
```

Comment:
redis/valkey replies with binary safe strings. So we should assume that our string is `&[u8]` or `Vec<u8>` and NOT a `String` (which has encoding etc).

Reply:
Agree
But I'd say "byte arrays" instead of "binary safe strings", because strings are always binary safe, but not vice versa.

Reply:
@eifrah-aws @Yury-Fridlyand I've refactored it as you said.